### PR TITLE
Improved Shooter code

### PR DESCRIPTION
**Pull request by Injector-Spenral**
The Shooter code now is in a controllers/components format. The Joystick controls have been changed slightly. The PID gains have been changed to those found in the characterisation. Robot.py's shooter control code has been cleaned up and moved to separate functions.

### CODE REVIEW

**Comment by auscompgeek** [comment]
what on earth are these merge commits in this PR

File: controllers/shooter_controller.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    def execute(self) -> None:
+        self.centre_pid.setReference(self.centre_rpm, rev.ControlType.kVelocity)
+        self.outer_pid.setReference(self.outer_rpm, rev.ControlType.kVelocity)
+
+    def setup(self) -> None:
+        self.configurePIDs()
+        self.shooter.loading_piston.setPulseDuration(1)
+
+    def configurePIDs(self) -> None:
+        self.centre_pid = self.shooter.centre_motor.getPIDController()
```
Comment (auscompgeek) [inline controllers/shooter_controller.py]: Why is this in the high-level controller? This should not be accessing implementation details of our components.

File: robot.py [hunk 2]
```diff
@@ -35,7 +37,7 @@ def createObjects(self):
         self.shooter_outer_motor = rev.CANSparkMax(3, rev.MotorType.kBrushless)
         self.shooter_centre_motor = rev.CANSparkMax(2, rev.MotorType.kBrushless)
 
-        self.loading_piston = wpilib.DoubleSolenoid(0, 1)
+        self.loading_piston = wpilib.Solenoid(0)
```
Comment (auscompgeek) [inline robot.py]: Please PR this change separately.

File: robot.py [hunk 3]
```diff
@@ -44,42 +46,37 @@ def createObjects(self):
...
-            self.loading_piston.set(wpilib.DoubleSolenoid.Value.kReverse)
-
+        self.handle_shooter_inputs(self.joystick_left)
         self.handle_indexer_inputs(self.joystick_left)
         self.handle_spinner_inputs(self.spinner_joystick)
-
-    def handle_indexer_inputs(self, joystick):
+        self.send_shooter_values()
+
+    def handle_shooter_inputs(self, joystick: wpilib.Joystick) -> None:
```
Comment (LucienMorey) [inline robot.py]: This level of abstraction seems vague. If you want to do it like this then maybe two methods for handling the piston and then handling the flywheels?

**Pull request closed without merging**